packages/db/index.js

```diff
@@ -457,25 +457,40 @@ export class DBClient {
     }
   }
 
+  /**
+   * @typedef {Object} listUploadReturn
+   * @property {number} count - The total number of uploads for that user (used for pagination)
+   * @property {Promise<Array<import('./db-client-types').UploadItemOutput>>} uploads - The list of requested uploads.
+   */
+
   /**
    * List uploads of a given user.
    *
    * @param {number} userId
    * @param {import('./db-client-types').ListUploadsOptions} [opts]
-   * @returns {Promise<Array<import('./db-client-types').UploadItemOutput>>}
+   * @returns {listUploadReturn}
    */
   async listUploads (userId, opts = {}) {
+    const offset = opts.offset || 0
+    const limit = offset + (opts.size || 25)
+
+    const sortBy = opts.sortBy === 'Name'
+      ? 'name'
+      : 'inserted_at'
+
+    const isAscendingSortOrder = opts.sortOrder === 'Asc'
+
     let query = this._client
       .from('upload')
-      .select(uploadQuery)
+      .select(uploadQuery, { count: 'exact' })
```

Comment:
I _think_ I need an exact count here, but unsure how this would impact the query performance, open to suggestion as to how I could improve this!

Reply:
This is definitively correct.
This could probably be optimised using `estimate`, however depending on how pagination has been implemented that might complicate a bit the logic.
So I'd monitor performance for big lists and update this if required.
What do you think?

Reply:
Yep, this makes sense - would be good to keep an eye on this when testing on staging. Thanks!